[Libs] Makes YML config file description more readable

pushkalkatara opened the pull request:
Instead of
```
Server: 1
 or
Server: 2
```
Changed to
```
Server: "Ice"
 or
Server: "ROS"
```

Solves issue https://github.com/JdeRobot/JdeRobot/issues/1060

Images to show working - 
![1](https://user-images.githubusercontent.com/21266230/36420845-3d87a1b6-162e-11e8-838f-e95fc2095087.png)
![3](https://user-images.githubusercontent.com/21266230/36420850-42d79c48-162e-11e8-961f-d91f678aa4e8.png)

pushkalkatara commented:
@fqez . Running Python file successfully loads the YML, but further due to 
```
sudo pip2 install zeroc-ice
```
i think pip installs wrapper for version 3.7 but JdeRobot supports 3.6. So after loading the image, error comes - 
![6](https://user-images.githubusercontent.com/21266230/36421038-ec3825be-162e-11e8-93b9-582530157eee.png)

fqez commented on src/libs/comm_py/comm/laserClient.py.only-ice.in:
```diff
@@ -66,8 +66,13 @@ def getLaserClient (jdrc, prefix):
+        server=1
+    elif server == "ROS":
+        server=2
+    else server = 0
```
you did miss a colon here ```else: server = 0```

fqez commented on src/libs/comm_py/comm/laserClient.py.ros.in:
```diff
@@ -75,8 +75,13 @@ def getLaserClient (jdrc, prefix):
+        server=1
+    elif server == "ROS":
+        server=2
+    else server = 0
```
you did miss a colon here ```else: server = 0```

Actually you missed that colon in every client, please, review it.

fqez commented on src/drivers/MAVLinkServer/MAVProxy/uav_viewer_py.yml:
```diff
@@ -1,30 +1,30 @@
 Camera:
-  Server: 1 # 0 -> Deactivate, 1 -> Ice , 2 -> ROS
+  Server: "Ice" # Deactivate, Ice , ROS
```
We prefer not to have quotes. As easy as possible for the end user :D ```Server: "Ice"``` --> ```Server: Ice```
Also, this should be case insensitive (ice, ICE, IcE, Ice, ROS, ros, RoS, rOS, ...)

Please, change this in every yml file!

fqez commented:
did you run ```pip install zeroc-ice```? If so, you maybe have installed ice 3.7  which we doesn't support (we support 3.6) so uninstall it and try to run ```sudo pip2 install zeroc-ice==3.6.3``` instead.

Also we need this feature integrated ASAP (other parts of the project depend on this particular one), so let me know if you can resolve it quickly. If not, I can merge this one in this current state and take care of the remaining problems myself.

Good job anyway! :)

pushkalkatara commented:
@fqez Thanks for the help. `pip install zeroc-ice==3.6.3` worked without any issues. How to update the documentation [here(jderobot-manual5)](http://jderobot.org/Manual-5)?
Also i have made the changes. Please review again. 
I've added - 
- [x] Removed the string colans. "ROS" ---> ros or "ICE" ---> ice
- [x] Now the yml supports every type-case. (ice, ICE, icE, Ice, ROS, ros, Ros, rOS etc)

Please review the changes.

fqez commented on src/libs/comm_py/comm/communicator.py.only-ice.in:
```diff
@@ -37,7 +37,7 @@ class Communicator:
 		ymlNode = self.config.getProperty(prefix)
 		for i in ymlNode:
-			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == 1:
+			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == "Ice":
```
You forgot the case sensitiviness here!

fqez commented on src/libs/comm_py/comm/communicator.py.ros.in:
```diff
@@ -38,9 +38,9 @@ class Communicator:
 		ymlNode = self.config.getProperty(prefix)
 		for i in ymlNode:
-			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == 1:
+			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == "Ice":
```
case sensitive!

fqez commented on src/libs/comm_py/comm/communicator.py.ros.in:
```diff
@@ -38,9 +38,9 @@ class Communicator:
+			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == "Ice":
 				iceserver = True
-			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == 2:
+			if type(ymlNode[i]) is dict and ymlNode[i]["Server"] == "ROS":
```
case sensitive

pushkalkatara commented:
@fqez Made the changes necessary changes. Good to go! :+1: 
Can you please help me with the pipeline of Python tests. It would help me debug the python comms.

fqez commented:
There is a conflict with  src/drivers/MAVLinkServer/MAVProxy/uav_viewer_py.yml so I can't merge this one automatically. Is it necessary? 
There are some test here https://github.com/JdeRobot/JdeRobot/tree/master/src/libs/comm_py/tests you can check them in order to test your modifications on the comm library.

Solve the conflict, please. :)

fqez commented:
Everything is now ok! Very nice job @pushkalkatara. We do appreciate your effort. Merging it, thanks!!

fqez commented:
undo the last commit please, you broke it! xD

pushkalkatara commented:
Sorry some git issues. Squashing my commits

pushkalkatara commented:
Thanks for the merge. :+1:
